File: RayTracerCs/LightSource.cs
using System.Collections.Generic;
using System.Linq;
using System.Numerics;

namespace RayTracerCs
{
    public class LightSource : TracerObject
    {
        public LightSource(
            Vector3 origin,
            Quaternion rotation,
            List<Triangle> triangles,
            float totalFlux,
            SpectralColor spectralColor
        ) : base(origin, rotation, triangles)
        {
            TotalFlux = totalFlux;
            Spectral = spectralColor.Normalize();
        }

        public float TotalFlux { get; }
        public SpectralColor Spectral { get; }

        public SpectralColor Luminance(float sqaredDistance)
        {
            return new SpectralColor(
                Spectral.Values.Select(val => val * TotalFlux / (sqaredDistance)).ToArray()
            );
        }
    }
}

File: RayTracer/LightSource.fs
module RayTracer.LightSource

open System.Numerics
open RayTracer.Color
open RayTracer.Geometry

type LightSource(origin: Vector3, rotation : Quaternion, triangles: Triangle array, totalFlux : float32, spectral : SpectralColor) =
    inherit TracerObject(origin, rotation, triangles)
    member this.flux: float32 = totalFlux
    member this.spectral: SpectralColor = spectral
